Gems/ROS2/Code/Include/ROS2/Camera/CameraPostProcessingRequestBus.h

```diff
@@ -24,21 +25,38 @@ namespace ROS2
     class CameraPostProcessingRequests : public AZ::EBusSharedDispatchTraits<CameraPostProcessingRequests>
     {
     public:
+        //! Each camera sensor component has its own post-processing bus.
         using BusIdType = AZ::EntityId;
         static constexpr AZ::EBusAddressPolicy AddressPolicy = AZ::EBusAddressPolicy::ById;
-        static constexpr AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Multiple;
+        //! Multiple post-processing functions can be registered to the bus.
+        //! They will be executed in the order
+        static constexpr AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::MultipleAndOrdered;
+
+        //! Priority of the post-processing bus.
+        //! @note higher priority buses will be processed first.
+        constexpr static AZ::u8 MIN_PRIORITY = 0;
+        constexpr static AZ::u8 MAX_PRIORITY = 255;
+        constexpr static AZ::u8 DEFAULT_PRIORITY = 127;
 
         //! Apply post-processing function, if any implementations to the bus are in the entity.
         //! @param image standard image message passed as a reference. It will be changed through post-processing.
-        //! @note you should check whether the encoding format is supported first with SupportsFormat.
+        //! @note Handler should always handle function call,
+        //! if handler does not support the format of the image, it should not change the image.
+        //! @note Every image is modified in place. Only one handler has access to the image at the time.
         virtual void ApplyPostProcessing(sensor_msgs::msg::Image& image) = 0;
 
-        //! Query whether a particular image encoding is supported by registered postprocessing.
-        //! @param encodingFormat name of the format.
-        virtual bool SupportsFormat(const AZStd::string& encodingFormat) = 0;
+        //! Get priority of the post-processing bus.
+        //! @return priority of the bus.
+        //! @note higher priority buses will be processed first.
+        virtual AZ::u8 GetPriority() const = 0;
 
-    protected:
-        ~CameraPostProcessingRequests() = default;
+        //! Compare two post-processing buses.
+        //! @param other bus to compare to.
+        //! @return true if this bus should be processed before the other.
+        inline bool Compare(const CameraPostProcessingRequests* other) const
```

Comment:
The name of this function is a little misleading to me. Its comparing priorities and returning of the current bus has a higher priority than the input, maybe rename to `IsHigherPriority` or `ComparePriority` ?

Reply:
Function with this convention is required for correct EBus ordering
According to `EBus.h`
```cpp
        /**
         * Sorting function for EBus event handlers.
         * Used only when the #HandlerPolicy is AZ::EBusHandlerPolicy::MultipleAndOrdered.
         * This function determines the order in which handlers at an address receive an event.
         *
         * By default, the function requires the handler to implement the following comparison
         * function.
         * @code{.cpp}
         * // Returns whether 'this' should precede 'other'.
         * bool Compare(const Interface* other) const;
         * @endcode
         */
        using BusHandlerOrderCompare = BusHandlerCompareDefault;

```

Reply:
Thanks for the clarification